src/group_theory/sylow.lean

```diff
@@ -600,6 +600,21 @@ begin
   exact P.1.card_subgroup_dvd_card,
 end
 
+/-- A subgroup with cardinality `p ^ n` is a Sylow group
+ where `n` is the multiplicity of `p` in the group order. -/
+def of_card [fintype G] {p : ℕ} [hp : fact p.prime] (H : subgroup G) [fintype H]
+  (card_eq: card H = p ^ (card G).factorization p) : sylow p G :=
+{ to_subgroup := H,
+  is_p_group' := is_p_group.of_card card_eq,
+  is_maximal' := begin
+    obtain ⟨P, hHP⟩ := (is_p_group.of_card card_eq).exists_le_sylow,
+    exact set_like.ext' (set.eq_of_subset_of_card_le hHP
+      (P.card_eq_multiplicity.trans card_eq.symm).le).symm ▸ λ _, P.3,
+  end }
+
+@[simp] lemma coe_of_card [fintype G] {p : ℕ} [hp : fact p.prime] (H : subgroup G) [fintype H]
+  (card_eq: card H = p ^ (card G).factorization p) : ↑(of_card H card_eq) = H := rfl
```

Comment:
```suggestion
  (card_eq) : ↑(of_card H card_eq) = H := rfl
```

Reply:
Hm, apparently this information is necessary to see that we use the same p as before, so I would keep it instead